xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -492,35 +488,33 @@ static void validateCommonTlsContext(
             "common-tls-context with tls_certificate_certificate_provider is not supported");
```

Comment:
Why throw on `hasTlsCertificateCertificateProvider()` only when `certInstanceName == null`?

Reply:
Good question! So `certInstanceName` is derived from the new field `tls_certificate_provider_instance` (or in its absence the old field `tls_certificate_certificate_provider_instance` which we will stop supporting as soon as the control plane makes the change). So if `certInstanceName == null` that means those fields are not set. And in such a situation if the control plane has populated the other fields that serve the same purpose (`tls_certificate_certificate_provider` or `tls_certificate_sds_secret_configs` or `tls_certificates`) then we have to generate this error.

However when `certInstanceName != null` then that means the field(s) we support is present and in that case even if those other fields are populated we don't complain. This way the control plane (TD) is able to send a payload that works for both gRPC and Envoy.